lib/hypertrack.dart

```diff
@@ -53,49 +81,82 @@ class HyperTrack {
   /// In case of frequent changes, it is possible, that intermediate states
   /// could be lost, so if more real-time responsiveness is required it is
   /// recommended to use [addGeotag] for passing that data.
-  void setDeviceMetadata(Map<String, Object> data) =>
-      _pluginInterface.setDeviceMetadata(data);
+  Future<void> setMetadata(Map<String, Object> data) {
+    return invokeSdkVoidMethod(SdkMethod.setDeviceMetadata, data);
+  }
 
-  /// Enable debugging.
-  void enableDebugLogging() async {
-    const MethodChannel('sdk.hypertrack.com/handle')
-        .invokeMethod('enableDebugLogging');
+  Future<Result<Location, TrackingError>> addGeotag(Map<String, Object> data) {
+    return invokeSdkMethod(SdkMethod.addGeotag, data);
+  }
+
+  Future<Result<Location, TrackingError>> get location async {
+    return invokeSdkMethod(SdkMethod.getLocation);
   }
 
   /// Triggers tracking start.
   ///
   /// This isn't always result in SDK tracking, as missing permissions or disabled
   /// geolocation sensors could lead to a tracking outage. Use [onTrackingStateChanged]
   /// stream to get the actual state details.
-  start() async {
-    await _pluginInterface.startTracking();
-  }
+  void startTracking() => invokeSdkVoidMethod(SdkMethod.startTracking);
 
   /// Stops tracking.
-  stop() async {
-    await _pluginInterface.stopTracking();
-  }
+  void stopTracking() => invokeSdkVoidMethod(SdkMethod.stopTracking);
 
   /// Allows you to use location mocking software (e.g. for development).
   ///
   /// Mock locations are ignored by HyperTrack SDK by default.
-  allowMockLocations([bool allow = false]) =>
-      _pluginInterface.allowMockLocations(allow);
+  void allowMockLocations() {
+    invokeSdkVoidMethod(SdkMethod.allowMockLocations, true);
+  }
 
-  /// Adds geotag with configurable payload.
-  ///
-  /// Please, bear in mind that this will be serialized as json so passing in
-  /// recursive data structure could lead to unpredictable results.
-  /// [expectedLocation] is a place, where action is supposed to occur.
-  /// Look [ExpectedLocation] class options for the details.
-  Future addGeotag(Map<String, Object> data,
-          [Location? expectedLocation]) async =>
-      _pluginInterface.addGeotag(data, expectedLocation);
+  void enableDebugLogging() {
```

Comment:
It's better to have `setLoggingEnabled(Bool)`

Reply:
Android SDK doesn't support setLoggingEnabled(false). I would suggest to leave it like this and fix with the new API.

Reply:
Why not making it right from the start, but it wouldn't work correctly with Android SDK in all permutations for now?

Reply:
I think it is worng approach. It's not hard to fix it later and it may confuse clients. The original goal of this task was to reduce discrepancies between platforms, and this will add more.

Reply:
the question is whether it's better to have an API breaking (minor version) release for deprecating this method signature, versus it being confusing that it doesn't "work" with `false` value?

also, i'm not convinced that the `false` parameter is ever passed to the function, imo @pavel-kuznetsov-hypertrack has this exact mindset as our developers would -- they just want to enable debugging. By default it's off, so disabling it is almost an afterthought. If anything I think they just comment out the method and re-run the build.